src/include/sof/audio/format_hifi3.h

```diff
@@ -27,4 +27,14 @@ static inline int16_t sat_int16(int32_t x)
 	return AE_SAT16X4(x, x);
 }
 
+static inline int8_t sat_int8(int32_t x)
+{
+	if (x > INT8_MAX)
+		return INT8_MAX;
+	else if (x < INT8_MIN)
+		return INT8_MIN;
+	else
+		return (int8_t)x;
```

Comment:
and intrinsic seems does not have special instruction to handle it, no sure about clamp instruction, @singalsu @andrula-song , do you have ideas to handle it? use sat16, then shift right 8?

anyway, in hifi3, use directly c seems not an efficient way.

Reply:
The HiFi3 & HiFi4 all contains a 16-entry 64-bit register file, so I am afraid there would not be a direct intrinsic for this.
"use sat16, then shift right 8?"
--> if for C code, I think this would be better than use sat16, and for HiFi version maybe use a 16-bit container left shift 8 bits with saturation and then right shift 8 bits.

Reply:
Ok, it seems Intrinsics dont make sense here. I think we are good then with generic C only.